src/tls.h

```diff
@@ -34,13 +34,10 @@
 
 struct enftun_tls
 {
-    struct enftun_tcp* sock; // the underlying TCP socket
-
-#ifdef USE_SCM
-    struct enftun_tcp_scm sock_scm;
-#else
-    struct enftun_tcp_native sock_native;
-#endif
+    struct enftun_tcp* sock;
+    enum enftun_tcp_type sock_type;
```

Comment:
No need to duplicate the `sock_type` here.   Too easy to get out of sync with the actual type, if we forget to copy/update at some point.

Instead of `tls->sock_type`, just use `tls->sock->type`.

Reply:
Requiring three variables (one for native, one for scm, and a pointer to the active one) feels ugly.  It complicates the `tls` code.  Let's try to hide that under some abstraction.  See my main comment on the PR.

Reply:
Done